File: frontend/src/types/accommodation.ts
export type AccommodationType = "hotel" | "guesthouse" | "homestay";

export interface AccommodationPhoto {
  id: string;
  url: string;
  public_id: string;
  caption: string | null;
}

export interface Accommodation {
  id: string;
  user_id: string;
  category: AccommodationType;
  name: string;
  location: string;
  latitude: number | null;
  longitude: number | null;
  price_range: string;
  amenities: string[];
  travel_tips: string | null;
  tags: string[];
  created_at: string;
  updated_at: string;
  photos: AccommodationPhoto[];
  author_name: string;
  author_picture_url: string | null;
  distance_km: number | null;
}

export interface AccommodationSearchParams {
  page?: number;
  per_page?: number;
  accommodation_type?: AccommodationType;
  price_range?: string;
  amenities?: string[];
  search?: string;
  sort_by?: "newest" | "name" | "price_asc" | "price_desc" | "distance";
  ref_lat?: number;
  ref_lng?: number;
}

export interface AccommodationListResponse {
  items: Accommodation[];
  total: number;
  page: number;
  per_page: number;
  total_pages: number;
}

export interface AIAccommodationRecommendation {
  accommodation_id: string;
  name: string;
  category: string;
  price_range: string;
  location: string;
  reasoning: string;
  estimated_cost_per_night: number;
  travel_convenience_score: number;
  cost_benefit_summary: string;
}

export interface AIRecommendationsResponse {
  recommendations: AIAccommodationRecommendation[];
  summary: string;
  itinerary_id: string;
}

File: backend/app/schemas/accommodation.py
import uuid
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict, Field


# ── Response schemas ──


class AccommodationPhotoRead(BaseModel):
    id: uuid.UUID
    url: str
    public_id: str
    caption: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class AccommodationRead(BaseModel):
    """Single accommodation (community entry of type hotel/guesthouse/homestay)."""

    id: uuid.UUID
    user_id: uuid.UUID
    category: str  # hotel, guesthouse, homestay
    name: str
    location: str
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    price_range: str
    amenities: List[str] = []
    travel_tips: Optional[str] = None
    tags: List[str] = []
    created_at: datetime
    updated_at: datetime

    photos: List[AccommodationPhotoRead] = []

    # Populated by service layer
    author_name: str
    author_picture_url: Optional[str] = None
    distance_km: Optional[float] = None  # distance from reference point

    model_config = ConfigDict(from_attributes=True)


class AccommodationListResponse(BaseModel):
    items: List[AccommodationRead]
    total: int
    page: int
    per_page: int
    total_pages: int


# ── AI recommendation schemas ──


class AIAccommodationRecommendation(BaseModel):
    """A single AI-generated accommodation recommendation."""

    accommodation_id: str
    name: str
    category: str
    price_range: str
    location: str
    reasoning: str  # LLM-generated explanation
    estimated_cost_per_night: float
    travel_convenience_score: int = Field(
        ..., ge=1, le=10, description="1-10 score for travel convenience"
    )
    cost_benefit_summary: str


class AIRecommendationsResponse(BaseModel):
    """Response containing AI-generated accommodation recommendations."""

    recommendations: List[AIAccommodationRecommendation]
    summary: str  # Overall LLM summary/reasoning
    itinerary_id: uuid.UUID
